presto-main/src/test/java/com/facebook/presto/block/AbstractTestBlock.java

```diff
@@ -49,10 +50,15 @@
 @Test
 public abstract class AbstractTestBlock
 {
-    protected <T> void assertBlock(Block block, T[] expectedValues)
+    protected <T> void assertBlock(Block block, BlockBuilder blockBuilderTemplate, T[] expectedValues)
```

Comment:
`Supplier<BlockBuilder> newBlockBuilder`

Reply:
@haozhun : And we pass in a lambda to generate new BlockBuilder? :)